main.go

```diff
@@ -106,11 +110,53 @@ func main() {
 	http.Handle("/live", handlers.NewLivenessHandler())
 	http.Handle("/metrics", promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
 
-	log.Info("Starting http handler...")
-	address := fmt.Sprintf(":%s", Options.ListenPort)
+	// Run listen on http and https ports if HTTPSCertFile/HTTPSKeyFile set
+	var httpsServer, httpServer http.Server
+
+	httpsListen := func() {
+		log.Infof("Starting https handler on %s...", httpsServer.Addr)
+		if err := httpsServer.ListenAndServeTLS(Options.HTTPSCertFile, Options.HTTPSKeyFile); err != http.ErrServerClosed {
+			log.Fatalf("HTTPS listener closed: %v", err)
+		}
+	}
+
+	httpListen := func() {
+		log.Infof("Starting http handler on %s...", httpServer.Addr)
+		if err := httpServer.ListenAndServe(); err != http.ErrServerClosed {
+			log.Fatalf("HTTP listener closed: %v", err)
+		}
+	}
+
+	// Interrupt servers on SIGINT/SIGTERM
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+
 	if Options.HTTPSKeyFile != "" && Options.HTTPSCertFile != "" {
-		log.Fatal(http.ListenAndServeTLS(address, Options.HTTPSCertFile, Options.HTTPSKeyFile, nil))
+		httpsServer = http.Server{
+			Addr: fmt.Sprintf(":%s", Options.ListenPort),
+		}
+		go httpsListen()
+		if Options.HTTPListenPort != "" {
+			httpServer = http.Server{
+				Addr: fmt.Sprintf(":%s", Options.HTTPListenPort),
+			}
+			go httpListen()
+		}
+	} else {
+		httpServer = http.Server{
+			Addr: fmt.Sprintf(":%s", Options.ListenPort),
+		}
+		go httpListen()
+	}
+	<-stop
```

Comment:
Should we have different behavior for TERM vs INT?
`http.Server` has `Close()` and `Shutdown()` for non-graceful/graceful respectively.
Do we think there's a reason to differentiate between the cases?

Reply:
Lets attempt `Close` when graceful didn't work (or took too long, but that might be too complicated)

Reply:
I suppose we could do a shutdown with a context, but yeah kube will just kill the container at some point so it may not be worth worrying about.